tests/apidoc_request_encoding_test.go

```diff
@@ -0,0 +1,29 @@
+
+package tests
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/matrix-org/complement/internal/b"
+	"github.com/matrix-org/complement/internal/match"
+	"github.com/matrix-org/complement/internal/must"
+)
+
+func TestRequestEncodingFails(t *testing.T) {
+	deployment := Deploy(t, "request_encoding", b.BlueprintAlice)
+	defer deployment.Destroy(t)
+	unauthedClient := deployment.Client(t, "hs1", "")
+	// sytest: POST rejects invalid utf-8 in JSON
+	t.Run("POST /login can login as user", func(t *testing.T) {
+		res := unauthedClient.MustDo(t, "POST", []string{"_matrix", "client", "r0", "login"}, json.RawMessage(`{
```

Comment:
Whoops, I think these were copied over from your other PR 🙂  The title should match the Sytest title, and the POST route should be "r0/register" rather than "r0/login".

Reply:
Ahh, my mistake here😓

Reply:
No worries 😄 I think the title should also be "POST rejects invalid utf-8 in JSON", right?

Reply:
Yes, sorry for the mistake again,I will be more cautious next time😅